examples/jsm/nodes/display/ColorSpaceNode.js

```diff
@@ -26,82 +35,75 @@ export const LinearTosRGB = new ShaderNode( ( inputs ) => {
 
 } );
 
-const EncodingLib = {
-	LinearToLinear,
-	LinearTosRGB
-};
+const getColorSpaceMethod = ( colorSpace ) => {
 
-class ColorSpaceNode extends TempNode {
+	let method = null;
 
-	constructor( method, node ) {
+	if ( colorSpace === LinearSRGBColorSpace ) {
 
-		super( 'vec4' );
+		method = 'Linear';
 
-		this.method = method;
+	} else if ( colorSpace === SRGBColorSpace ) {
 
-		this.node = node;
+		method = 'sRGB';
 
 	}
 
-	fromColorSpace( colorSpace ) {
-
-		let method = null;
-
-		if ( colorSpace === LinearSRGBColorSpace ) {
-
-			method = 'Linear';
-
-		} else if ( colorSpace === SRGBColorSpace ) {
-
-			method = 'sRGB';
+	return method;
 
-		}
-
-		this.method = 'LinearTo' + method;
-
-		return this;
-
-	}
-
-	fromEncoding( encoding ) { // @deprecated, r152
-
-		console.warn( 'THREE.ColorSpaceNode: Method .fromEncoding renamed to .fromColorSpace.' );
+};
 
-		let method = null;
+const getMethod = ( source, target ) => {
 
-		if ( encoding === LinearEncoding ) {
+	return getColorSpaceMethod( source ) + 'To' + getColorSpaceMethod( target );
 
-			method = 'Linear';
+};
 
-		} else if ( encoding === sRGBEncoding ) {
+class ColorSpaceNode extends TempNode {
 
-			method = 'sRGB';
+	constructor( method, node ) {
 
-		}
+		super( 'vec4' );
 
-		this.method = 'LinearTo' + method;
+		this.method = method;
 
-		return this;
+		this.node = node;
 
 	}
 
 	construct() {
 
 		const { method, node } = this;
 
-		return EncodingLib[ method ].call( { value: node } );
+		if ( method === ColorSpaceNode.LINEAR_TO_LINEAR )
+			return node;
+
+		return Methods[ method ].call( { value: node } );
 
 	}
 
 }
 
 ColorSpaceNode.LINEAR_TO_LINEAR = 'LinearToLinear';
-ColorSpaceNode.LINEAR_TO_SRGB = 'LinearTosRGB';
+ColorSpaceNode.LINEAR_TO_sRGB = 'LinearTosRGB';
+ColorSpaceNode.sRGB_TO_LINEAR = 'sRGBToLinear';
+
+const Methods = {
+	[ ColorSpaceNode.LINEAR_TO_sRGB ]: LinearTosRGBShader,
+	[ ColorSpaceNode.sRGB_TO_LINEAR ]: sRGBToLinearShader
+};
 
 export default ColorSpaceNode;
 
-export const colorSpace = ( node, colorSpace ) => nodeObject( new ColorSpaceNode( null, nodeObject( node ) ).fromColorSpace( colorSpace ) );
+export const linearToColorSpace = ( node, colorSpace ) => nodeObject( new ColorSpaceNode( getMethod( LinearSRGBColorSpace, colorSpace ), nodeObject( node ) ) );
```

Comment:
Maybe just `toColorSpace` and `fromColorSpace`?

Reply:
I think it's is not possible currently. We need to know what the source is to made the conversion properly.